assets/components/src/autocomplete-tokenfield/index.js

```diff
@@ -0,0 +1,184 @@
+		const { validValues } = this.state;
+		return values.reduce(
+			( accumulator, value ) =>
+				validValues[ value ] ? [ ...accumulator, validValues[ value ] ] : accumulator,
+			[]
+		);
+	}
+
+	/**
+	 * Get a list of values for input labels.
+	 *
+	 * @param {Array} labels Array of labels from the tokens.
+	 * @return {Array} Array of valid values corresponding to the labels.
+	 */
+	getValuesForLabels( labels ) {
+		const { validValues } = this.state;
+		return labels.map( label =>
+			Object.keys( validValues ).find( key => validValues[ key ] === label )
+		);
+	}
+
+	/**
+	 * Refresh the autocomplete dropdown.
+	 *
+	 * @param {string} input Input to fetch suggestions for
+	 */
+	updateSuggestions( input ) {
+		const { fetchSuggestions } = this.props;
+		if ( ! fetchSuggestions ) {
+			return;
+		}
+
+		this.setState( { loading: true }, () => {
+			const request = fetchSuggestions( input );
+			request
+				.then( suggestions => {
+					// A fetch Promise doesn't have an abort option. It's mimicked by
+					// comparing the request reference in on the instance, which is
+					// reset or deleted on subsequent requests or unmounting.
+					if ( this.suggestionsRequest !== request ) {
+						return;
+					}
+
+					const { validValues } = this.state;
+					const currentSuggestions = [];
+
+					suggestions.forEach( suggestion => {
+						currentSuggestions.push( suggestion.label );
+						validValues[ suggestion.value ] = suggestion.label;
+					} );
+
+					this.setState( { suggestions: currentSuggestions, validValues, loading: false } );
+				} )
+				.catch( () => {
+					if ( this.suggestionsRequest === request ) {
+						this.setState( {
+							loading: false,
+						} );
+					}
+				} );
+
+			this.suggestionsRequest = request;
+		} );
+	}
+
+	/**
+	 * When a token is selected, we need to convert the string label into a recognized value suitable for saving as an attribute.
+	 *
+	 * @param {Array} tokenStrings An array of token label strings.
+	 */
+	handleOnChange( tokenStrings ) {
+		const { onChange } = this.props;
+		onChange( this.getValuesForLabels( tokenStrings ) );
+	}
+
+	/**
+	 * To populate the tokens, we need to convert the values into a human-readable label.
+	 *
+	 * @return {Array} An array of token label strings.
+	 */
+	getTokens() {
+		const { tokens } = this.props;
+		return this.getLabelsForValues( tokens );
+	}
+
+	/**
+	 * Render.
+	 */
+	render() {
+		const { help, label = '', maxLength } = this.props;
+		const { suggestions, loading } = this.state;
+
+		return (
+			<div className="newspack-autocomplete-tokenfield">
+				<FormTokenField
+					value={ this.getTokens() }
+					suggestions={ suggestions }
+					onChange={ tokens => this.handleOnChange( tokens ) }
+					onInputChange={ input => this.debouncedUpdateSuggestions( input ) }
+					label={ label }
+					maxLength={ maxLength }
+				/>
+				{ loading && <Spinner /> }
+				{ help && <p className="autocomplete-tokenfield__help">{ help }</p> }
```

Comment:
The classname should also be updated here ;)

Reply:
Doh!